### PR TITLE
Allow user to opt out of edge attributes in from_numpy_array

### LINKED ISSUE #6248: Allow caller to opt out of adding `weight` attributes in `from_numpy_array`
### Current Behavior

Right now, calls to `from_numpy_array` add a `weight` attribute to all edges in the newly created (non-multi)graph. 

### Expected Behavior

It would be nice to allow the user to opt out of this behavior, or to specify an attribute that should receive this weight.

### Steps to Reproduce

```python
import networkx as nx
import numpy as np

mat = np.array([
    [0, 1],
    [1, 0]
])
g = nx.from_numpy_array(mat, create_using=nx.DiGraph)
print(g.edges(data=True))
```

```
[(0, 1, {'weight': 1}), (1, 0, {'weight': 1})]
```

### Environment

* Python version: 3.10
* NetworkX version: All (2.6.3)

### Additional context

This is confusing behavior because the caller does not explicitly request attribute creation. I would propose the following PREFERRED behavior:

#### Preferred

Creating a graph with an array of `dtype` ∈ {int, float} creates an attribute of name `weight_attribute`, which defaults to `w` for backwards compat.

```python
g = nx.from_numpy_array(mat, create_using=nx.DiGraph, weight_attribute="w")
print(g.edges(data=True))
```

```
[(0, 1, {'w': 1}), (1, 0, {'w': 1})]
```

```

Creating a graph with an array of `dtype` = bool treats each value as an existence flag and does not create an attribute:


```python
g = nx.from_numpy_array(mat==1, create_using=nx.DiGraph, weight_attribute="w")
print(g.edges(data=True))
```

```
[(0, 1, {}), (1, 0, {})]
```

#### Also seems fine but changes fn signature:

Perhaps better for backwards compat is to allow the user to opt-out of the current behavior with a new argument `weight_attribute` which defaults to `weight` but can be set to `None` to remove this behavior. (Perhaps a bit more predictable than the above option?)

#### Seems kinda jank but is totally back-compat:

Setting `parallel_edges=True` and `create_using=nx.DiGraph` currently doesn't do anything, and could be repurposed to perform this behavior. Eh. This is kinda gross.

### PR DESCRIPTION
**Pull request by j6k4m8**
Begins to resolve #6248; tagging so @rossbar and @dschult can follow along :)

- [x] Update function signature and docstring
- [x] Verify existing tests to ensure new behavior is backwards compatible
- [x] Add new tests for new behavior

### CODE REVIEW

**Review by dschult — APPROVED** [review]
This looks good!  Thanks!

**Review by rossbar — COMMENTED** [review]
Generally LGTM, just a couple high-level comments!

File: networkx/tests/test_convert_numpy.py [hunk 1]
```diff
@@ -150,6 +150,32 @@ def test_from_numpy_array_parallel_edges(self):
         )
         assert graphs_equal(actual, expected)
 
+    @pytest.mark.parametrize(
+        "dt",
+        (
+            None,  # default
+            int,  # integer dtype
+            np.dtype([("weight", "f8"), ("color", "i1")]),  # Structured dtype with named fields
```
Comment (SultanOrazbayev) [inline networkx/tests/test_convert_numpy.py:158]:
```suggestion
            np.dtype(
                [("weight", "f8"), ("color", "i1")]
            ),  # Structured dtype with named fields
```

File: networkx/convert_matrix.py [hunk 2]
```diff
@@ -1019,6 +1019,10 @@ def from_numpy_array(A, parallel_edges=False, create_using=None):
     create_using : NetworkX graph constructor, optional (default=nx.Graph)
        Graph type to create. If graph instance, then cleared before populated.
 
+    edge_attr : String, optional (default="weight")
+        The attribute to which the array values are assigned on each edge. If
+        it is Falsy, edge attributes will not be assigned.
```
[the remark this answers is not in the document]
Comment (j6k4m8) [reply networkx/convert_matrix.py]: Fixed in the latest!

File: networkx/tests/test_convert_numpy.py (hunk 1 above)
Comment (j6k4m8) [reply networkx/tests/test_convert_numpy.py]: Thanks @SultanOrazbayev  :)

**Comment by rossbar** [comment]
The currently failing test (as of b9218fb) illustrates an interesting corner case: what happens with `edge_attr=None` (indicating the user doesn't want any attributes) but the dtype of the array is structured (which is currently interpreted as "map all named attributes to edge attributes in the graph").

I *think* what we want is for `edge_attr` to override the `dtype` - in which case it looks like the code may need to be adjusted to handle this case.

**Comment by j6k4m8** [comment]
Indeed, I was wodnering about how best to handle this! Very nice to have such great unit tests on this repo :) My PhD qualifying exam is tomorrow so I'm unlikely to hit this before thursday (eek!) but I will address asap! Thank you for the guidance!

**Review by MridulS — APPROVED** [review]
Thanks @j6k4m8 !

File: networkx/convert_matrix.py [hunk 3]
```diff
@@ -1127,7 +1136,9 @@ def from_numpy_array(A, parallel_edges=False, create_using=None):
                 {
                     name: kind_to_python_type[dtype.kind](val)
                     for (_, dtype, name), val in zip(fields, A[u, v])
-                },
+                }
+                if edge_attr
```
Comment (dschult) [inline networkx/convert_matrix.py:1140]: This `if` statement will result in an empty dict for many different values that evaluate to `False`. The docs mention False and None, but others like `0`, `""` and any empty container would act like `None`.  Would it be better to check e.g. `if edge_attr is None`?

Comment (j6k4m8) [reply networkx/convert_matrix.py:1140]: That's a good point — I'll open a new PR as this one is merged!

File: networkx/convert_matrix.py [hunk 4]
```diff
@@ -1034,6 +1038,11 @@ def from_numpy_array(A, parallel_edges=False, create_using=None):
     indicated by the upper triangle of the array `A` will be added to the
     graph.
 
+    If `edge_attr` is Falsy (False or None), edge attributes will not be
```
Comment (MridulS) [inline networkx/convert_matrix.py:1041]: Hmm, should we change this then too?

0 is Falsy (I think?)
